.pre-commit-config.yaml

```diff
@@ -0,0 +1,59 @@
+# Pre-commit is tool which allows to run checks automatically before commit.
+# This way commits are always consistent and clean.
+
+exclude: target/|\.git/|.*\.snap
+
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: c4a0b883114b00d8d76b479c820ce7950211c99b # v4.5.0
+    hooks:
+      - id: check-merge-conflict
+        args: [--assume-in-merge]
+      # Editorconfig-checker does not care if file contains multiple newlines.
+      - id: end-of-file-fixer
+      - id: check-yaml
+      - id: check-case-conflict
+      - id: trailing-whitespace
+      - id: fix-byte-order-marker
+
+  - repo: https://github.com/Lucas-C/pre-commit-hooks
+    rev: 762c66ea96843b54b936fc680162ea67f85ec2d7 # v1.5.4
+    hooks:
+      - id: remove-tabs
+        name: Remove tabs (4 spaces)
+        args: ['--whitespaces-count', '4']
+        exclude: \.(yaml|yml)$
+      - id: remove-tabs
+        name: Remove tabs (2 spaces)
+        args: ['--whitespaces-count', '2']
+        files: \.(yaml|yml)$
+
+  - repo: https://github.com/pre-commit/pygrep-hooks
+    rev: 3a6eb0fadf60b3cccfd80bad9dbb6fae7e47b316 # v1.10.0
+    hooks:
+      - id: text-unicode-replacement-char
+        exclude: ".*test_fdx_utf8.cpp$"
+
+  # Formatting Python code. We have not so many files so no example flake8 yet.
```

Comment:
~Not sure why we would need a python formatter~
Looks like we have a single python file to generate datatypes in the stdlib.
I don't think it would hurt to keep but we could also just remove that script later on, it's no longer really needed.

Reply:
Yeah just happit of mine to try get every possible files to have formatter. It does not hurt anything as this only applies when it is python file. I can take it out if wanted.

Reply:
No need to remove it, if we end up removing the python part we'll just also remove the formatter at that time.